### PR TITLE
Add sign.builds projects to facilitate real signing in official builds.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
New build project, sign.builds, is to be built post building of binaries
so that they can be real-signed in the official build.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@weshaggard Signing implementation for CoreCLR

File: src/sign.builds [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  <ItemDefinitionGroup>
+    <FilesToSign>
+      <Authenticode>$(AuthenticodeSig)</Authenticode>
+      <StrongName>$(StrongNameSig)</StrongName>
+    </FilesToSign>
+  </ItemDefinitionGroup>
+
+  <!-- gather the list of binaries to sign with the default certificates -->
+  <ItemGroup>
+    <FilesToSign Include="$(BinDir)*.dll" Exclude="$(BinDir)*.ni.dll" />
```
Comment (weshaggard) [inline src/sign.builds]: Don't we need to Authenticode sign ni's as well?

File: src/sign.builds [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    <FilesToSign Include="$(BinDir)*.exe" />
+  </ItemGroup>
+
+  <!--
+    for some reason the signing task incorrectly attemps to strong-name sign
+    native images which causes the signing step to fail for obvious reasons.
+  -->
+  <ItemGroup>
+    <FilesToSign Include="$(BinDir)*.ni.dll">
```
Comment (weshaggard) [inline src/sign.builds]: Doesn't this also happen for other native .dll's as well? Or does the sign task correctly detect those?

Comment (jhendrixMSFT) [inline src/sign.builds]: It correctly detects the rest.  Whatever algorithm they're using appears to be broken for NIs.

**Comment by weshaggard** [comment]
I assume you are doing this for everything instead of using the sign marker file because we build a lot of native projects which doesn't have a good way to drop the marker file. What is the plan for exclude files that we don't want to be signed?

**Comment by jhendrixMSFT** [comment]
@weshaggard correct, the use of cmake precludes us from using the same mechanism as in CoreFX.
As for excludes, if we use this current pattern one would have to explicitly opt out of signing.  How common is this requirement (I would think it rare but it's just a guess)?  We could flip the pattern to make it opt in however one must remember to update sign.builds whenever a new binary file is added (also rare although I would think it more common that producing a binary that we don't want to sign, not to mention the troubles we've had with unsigned files in the past).

**Comment by jhendrixMSFT** [comment]
@weshaggard If you have no other concerns I'd like to get this merged today to unblock the official build pipeline.

**Comment by weshaggard** [comment]
LGTM

File: src/mscorlib/ref/mscorlib.csproj [hunk 3]
```diff
@@ -68,6 +68,10 @@
 
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.Targets" />
 
+  <PropertyGroup>
+    <StrongNameSig>Silverlight</StrongNameSig>
```
Comment (weshaggard) [inline src/mscorlib/ref/mscorlib.csproj]: Is this actually being used? Is it getting overrriden in sign.builds?

Comment (jhendrixMSFT) [inline src/mscorlib/ref/mscorlib.csproj]: Yes when the reference assembly is generated.

**Pull request merged**